### DISCUSSION TITLE
Welcome to Keras Discussions!

### DISCUSSION THREAD

**Discussion by fchollet** [keras-team/keras#19700] [Announcements]
## 👋 Welcome!

Discussions is a place for you to connect with other members of the Keras community.

We hope that you:

* Ask your own support questions and answer the questions posted by others. Note that Keras maintainers are not actively monitoring questions here.
* Share Keras-related projects and information.
* Engage with other community members in productive discussions.
* Welcome others and are open-minded. Remember that this is a community we
build together 💪

Do not:

* Post bug reports. You can use GitHub Issues for that instead.
* Make feature requests. You can use GitHub Issues for that instead.

To get started, you can comment below with a brief self-introduction and description of how you use Keras.

**Comment by prthmptl** [comment]
I want some ideas on how can I add recommendations for metal health problems with very little data. Me and my friend working on this project and he already added content based recommendation but our mentor keeps telling us that add something more like in recommendations but the thing is we don't have that much data. So, what can I do to add another approach of recommendation which requires less data.

**Comment by ghost** [comment]
I use Keras extensively for its intuitive API and powerful deep learning capabilities.  Currently, I'm applying it to my project in **time-series analysis**, building a predictive maintenance model for industrial equipment. I'm always looking for new ways to optimize my models and learn best practices.

**Comment by alexx96ew** [comment]
Ok

**Comment by roebel** [comment]
Hello, thanks for the initiative. 

I am working on audio signal processing, mostly voice but also music. 
Here an example: https://github.com/roebel/MBExWN_Vocoder.
I started with Tensorflow 1, than with TF2 I started using keras, and now I am preparing to switch to keras 3.

While I like the high-level API and encapsulation, and I tend to subclass models and layers for all I do.
I am sometimes a bit frustrated about the incomplete documentation.
Following these discussions, I would hope to gather more insights into the underlying philosophy and strategies 
of Models and layers.

As an example, I struggled a while with the interface of Layer.add_metric until I finally decided to handle all the metrics in my CutomLayers myself. At some point, I forgot to implement the related property (Layer.metrics) and I was astonished to see that the model had tracked my metrics anyway.

Just a few days ago I again had the question of how model outputs, losses and metrics, and finally target values will be paired. 
I have often wondered about the possibility of using dicts to match metrics with output values, but for the moment I have not seen how I can produce named outputs in a custom model without using the Functional API.

I posed a question about a specific problem on [StackOverflow](https://stackoverflow.com/questions/78536202/how-to-train-a-keras-model-with-a-single-output-using-two-different-metrics-with?noredirect=1#comment138459850_78536202). #19386 
For the moment I believe that there is no elegant solution for this situation, and so I best overwrite train_step and test_step. But, given I do not fully understand the dict-based mapping, this may not be correct. If there are suggestions I'd be happy to discuss these.
